[MapDB] should just create a default directory and data store if it doesn't already exist

<!--- Provide a general summary of the issue in the Title above -->

<!--- If the issue is related to a binding, please include it in the title. -->

<!--- For example, "[astro] My issue..." -->

<!--- For questions or requests for assistance please use community.openhab.org instead. -->

<!--- Feel free to delete any comment lines in the template (starting with &lt;!--) -->
## Expected Behavior

<!--- If you're describing a bug, tell us what should happen -->

<!--- If you're suggesting a change/improvement, tell us how it should work -->

When using MapDB on a new openHAB 2 install it should create missing parent folders for the path to its persistence files instead of throwing an error in the logs.
## Current Behavior

<!--- If describing a bug, tell us what happens instead of the expected behavior -->

<!--- When possible, include related log information (preferably debug level) and related configs. -->

<!--- Use file attachment for log and config information longer than a few lines -->

<!--- If suggesting a change/improvement, explain the difference from current behavior -->

<!--- For improvements, discuss in at community.openhab.org first and include link to topic -->

When migrating to a brand new openHAB 2 install and first starting up MapDB persistence an error gets thrown in the logs complaining about the parent directory not existing.

`2016-09-24 12:14:59.335 [ERROR] [org.openhab.persistence.mapdb       ] - [org.openhab.persistence.mapdb(194)] The activate method has thrown an exception
java.io.IOError: java.io.IOException: Parent folder does not exist: /opt/openhab2/userdata/persistence/mapdb/storage.mapdb`

To correct the error one need only manually create the persistence/mapdb folder.

This may primarily be a problem with openHAB 2 manual installs only as I saw it when setting up openHAB 2 in Docker and [this thread](https://community.openhab.org/t/oh2-snapshot-online-version-using-wrong-folder-for-mapdb-persistence/14434) is using paths that imply a manual install.
## Possible Solution

<!--- Not obligatory, but suggest a fix/reason for the bug, -->

<!--- or ideas how to implement the addition or change -->

Have the binding create missing parent directories if they are missing.
## Steps to Reproduce (for bugs)

<!--- Provide a link to a live example, or an unambiguous set of steps to -->

<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Stop openHAB 2
2. Move or remove the userdata/persistence/mapdb folder
3. Start openHAB 2
## Context

<!--- How has this issue affected you? What are you trying to accomplish? -->

<!--- Providing context helps us come up with a solution that is most useful in the real world -->

Initial setup and configuration of openHAB 2. It was an unexpected, if easily manually mitigated error.
## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->
- Version used: openHAB 2, latest snapshot on Docker image, linked to thread above is using some openHAB 2 apparently manually installed
- Official Docker Image, not sure about linked to thread
- Ubuntu 16.1 based OS
##

--- a/bundles/persistence/org.openhab.persistence.mapdb/src/main/java/org/openhab/persistence/mapdb/internal/MapDBPersistenceService.java
+++ b/bundles/persistence/org.openhab.persistence.mapdb/src/main/java/org/openhab/persistence/mapdb/internal/MapDBPersistenceService.java
@@ -81,7 +81,7 @@ public class MapDBPersistenceService implements QueryablePersistenceService {
     private static Map<String, MapDBItem> map;
 
     public void activate(final BundleContext bundleContext, final Map<String, Object> config) {
-        logger.debug("mapdb persistence service activated");
+        logger.debug("mapdb persistence service is being activated");
 
         String commitIntervalString = (String) config.get("commitinterval");
         if (StringUtils.isNotBlank(commitIntervalString)) {
@@ -102,7 +102,11 @@ public class MapDBPersistenceService implements QueryablePersistenceService {
 
         File folder = new File(DB_FOLDER_NAME);
         if (!folder.exists()) {
-            folder.mkdir();
+            if (!folder.mkdirs()) {
+              logger.error("Failed to create one or more directories in the path '{}'", DB_FOLDER_NAME);
+              logger.error("MapDB persistence service activation has failed.");
+              return;
+            }
         }
 
         File dbFile = new File(DB_FOLDER_NAME, DB_FILE_NAME);
@@ -110,6 +114,7 @@ public class MapDBPersistenceService implements QueryablePersistenceService {
         Serializer<MapDBItem> serializer = new MapDBitemSerializer();
         map = db.createTreeMap("itemStore").valueSerializer(serializer).makeOrGet();
         scheduleJob();
+        logger.debug("mapdb persistence service is now activated");
     }
 
     public void deactivate(final int reason) {

